mkosi/__init__.py

```diff
@@ -677,113 +650,82 @@ def resolve_module_dependencies(root: Path, kver: str, modules: Sequence[str]) -
             key, sep, value = line.partition("=")
 
         if key in ("depends", "softdep"):
-            for d in value.strip().split(","):
-                if not d:
-                    continue
-
-                if d not in nametofile:
-                    logging.warning(f"{d} is a dependency of {name} but is not installed, ignoring ")
-                    continue
-
-                depends.append(d)
+            depends += [d for d in value.strip().split(",") if d]
 
         if key == "firmware":
-            for f in root.joinpath("usr/lib/firmware").glob(f"{value.strip()}*"):
-                firmware.append(f)
-
-        if key == "filename":
-            if name:
-                moddep[name] = depends
-                firmwaredep[name] = firmware
+            firmware += [f.relative_to(root) for f in root.joinpath("usr/lib/firmware").glob(f"{value.strip()}*")]
 
-                depends = []
-                firmware = []
+        if key == "name":
+            name = value.strip()
 
-            name = module_path_to_name(Path(value))
+            moddep[name] = depends
+            firmwaredep[name] = firmware
 
-    # Make sure we add the last module as well.
-    moddep[name] = depends
-    firmwaredep[name] = firmware
+            depends = []
+            firmware = []
 
-    todo = [*modules]
+    todo = [*builtin, *modules]
     mods = set()
     firmware = set()
 
     while len(todo) > 0:
```

Comment:
```suggestion
    while todo:
```
Since the empty list is `False`, this is a bit more pythonic (also more C, I guess :) )